core/src/main/java/org/jdbi/v3/core/extension/Extensions.java

```diff
@@ -89,8 +90,34 @@ public boolean hasExtensionFor(Class<?> extensionType) {
                 .findFirst();
     }
 
+    /**
+     * Allow using {@link java.lang.reflect.Proxy} to implement extensions.
+     * @param allowProxy whether to allow use of Proxy types
+     * @return this
+     */
+    public Extensions setAllowProxy(boolean allowProxy) {
```

Comment:
Not that I have a better idea, but have you thought it through that presenting this feature as a boolean decision "proxy or other" is the way to go? Not something like `setSqlObjectFactory` or something where you can plug in a `ProxyFactory` and a `GeneratedClassFactory`? A distinction like that might also be a good starting point for later refactoring, since as far as I understand the point is to have two mutually exclusive sqlobject generators, so slapping their code together in 1 class sounds like conflation.

Reply:
Well, the point of this toggle really is, "inform Jdbi that it is forbidden to make Proxy objects" -- usually because it's unimplemented or not desired.  This is not meant to be a general "switch between SqlObject" configuration toggle.

Reply:
Ok, so more of a general setting to be used by potential other features too, just to indicate whether or not Proxies are a go.